### PR TITLE
Filter button shows full list if no filter is selected in.

### PR DESCRIPTION
**Pull request by ayush290104**
#1397 
This pull request is in order to solve issue 1397 and from now selecting no filter gives the full list of accounts.

### CODE REVIEW

File: app/src/main/java/org/mifos/mobile/ui/fragments/AccountsFragment.kt [hunk 1]
```diff
@@ -363,14 +363,21 @@ class AccountsFragment : BaseFragment(), OnRefreshListener, AccountsView {
      */
     fun filterSavingsAccount(statusModelList: List<CheckboxStatus?>?) {
         val filteredSavings: MutableList<SavingAccount?> = ArrayList()
-        if (accountsPresenter?.getCheckedStatus(statusModelList) != null && accountsPresenter != null) {
+        if (accountsPresenter?.getCheckedStatus(statusModelList) != null && accountsPresenter != null&&accountsPresenter?.getCheckedStatus(statusModelList)!!
+                .isNotEmpty()) {
```
Comment (PratyushSingh07) [inline app/src/main/java/org/mifos/mobile/ui/fragments/AccountsFragment.kt:367]: use safe call operator instead of not-null assertion operator

**Review by PratyushSingh07 — CHANGES_REQUESTED** [review]

**Comment by PratyushSingh07** [comment]
you also need to create anothe branch , dont push the changes from the default branch . 
use `git checkout -b <branch-name>` to create a new branch 

It will be better if you could attach a video to demonstrate the changes

File: app/src/main/java/org/mifos/mobile/ui/fragments/AccountsFragment.kt (hunk 1 above)
Comment (ayush290104) [reply app/src/main/java/org/mifos/mobile/ui/fragments/AccountsFragment.kt:367]: Already we checked for null at the starting of the if statement so we can use a not null assertion.

**Comment by ayush290104** [comment]
@PratyushSingh07 sir should i close this pr and and push from other branch in a different pr

File: app/src/main/java/org/mifos/mobile/ui/fragments/AccountsFragment.kt (hunk 1 above)
Comment (PratyushSingh07) [reply app/src/main/java/org/mifos/mobile/ui/fragments/AccountsFragment.kt:367]: you are right but using the safe call operator again provides an additional layer of null safety, ensuring that the subsequent method call is only executed if accountsPresenter is not null.

**Pull request closed without merging**